src/index.ts

```diff
@@ -37,13 +39,31 @@ async function installCli(version: string, nodePath: string) {
   await exec.exec(installCommand, [], options);
 }
 
-function configureWorkspace(workspace: string) {
-  exec.exec(`mabl config set workspace ${workspace} && mabl config list`);
+async function configureWorkspace(
+  workspace: string,
+  nodePath: string,
+): Promise<boolean> {
+  const options = {
+    cwd: nodePath,
+  };
+
+  try {
+    await exec.exec(`mabl config set workspace ${workspace}`, [], options);
+  } catch (err) {
+    core.setFailed(
+      `Failed while trying to configure workspace with error ${err}`,
+    );
+
+    return false;
+  }
+  await exec.exec(`mabl config list`, [], options);
+
+  return true;
 }
 
-function findNode() {
-  const allNodeVersions = toolCache.findAllVersions('node');
-  if (!allNodeVersions) {
+async function findNode() {
+  const allNodeVersions = await toolCache.findAllVersions('node');
+  if (!allNodeVersions || !allNodeVersions[0]) {
```

Comment:
Possible bang reduction: 
```suggestion
  if (!(allNodeVersions && allNodeVersions[0])) {
```